### PR TITLE
Explicitly specify command names with underscores (continued)

### PR DESCRIPTION
**Pull request by jleveque**
Starting with Click 7.0, the default behavior of a command with underscores will be to replace with dashes.

This is to address the above default behavior change, so that the commands remain the same, extending the work done in https://github.com/Azure/sonic-utilities/pull/846 to all Click-based SONiC command line utilities.

Resolves https://github.com/Azure/sonic-utilities/issues/847

More details can be found here: https://github.com/pallets/click/issues/1123

For new subcommands which haven't been included in a release branch yet, use hyphens, as this will be the direction we move in the future. I have also opened an issue to track the eventual transition: https://github.com/Azure/sonic-utilities/issues/853

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -1375,7 +1375,7 @@ def memory(kdump_memory):
         config_db.mod_entry("KDUMP", "config", {"memory": kdump_memory})
         run_command("sonic-kdump-config --memory %s" % kdump_memory)
 
-@kdump.command()
+@kdump.command('num_dumps')
```
Comment (lguohan) [inline config/main.py]: this command hasn't been released. so, i suggest change to num-dumps

File: config/main.py [hunk 2]
```diff
@@ -2092,7 +2092,7 @@ def delete(counter_name, verbose):
 #
 # 'add_reasons' subcommand ('config dropcounters add_reasons')
 #
-@dropcounters.command()
+@dropcounters.command('add_reasons')
```
Comment (lguohan) [inline config/main.py]: is this release as well? I suggest to change to add-reasons

File: config/main.py [hunk 3]
```diff
@@ -2105,7 +2105,7 @@ def add_reasons(counter_name, reasons, verbose):
 #
 # 'remove_reasons' subcommand ('config dropcounters remove_reasons')
 #
-@dropcounters.command()
+@dropcounters.command('remove_reasons')
```
Comment (lguohan) [inline config/main.py]: is this release as well? I suggest to change to add-reasons

File: debug/main.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def bgp():
         """debug bgp group """
         pass
 
-    @bgp.command()
+    @bgp.command('allow_martians')
```
Comment (lguohan) [inline debug/main.py]: allow_martians -> allow-martians

File: debug/main.py [hunk 5]
```diff
@@ -71,7 +71,7 @@ def keepalives(prefix_or_iface):
         command += '"'
         run_command(command)
 
-    @bgp.command()
+    @bgp.command('neighbor_events')
```
Comment (lguohan) [inline debug/main.py]: we should keep use - instead of underscore.

File: pddf_fanutil/main.py [hunk 6]
```diff
@@ -178,7 +178,7 @@ def debug():
     """pddf_fanutil debug commands"""
     pass
 
-@debug.command()
+@debug.command('dump_sysfs')
```
Comment (lguohan) [inline pddf_fanutil/main.py]: this hasn't release in 201911 i think.

File: config/main.py (hunk 1 above)
Comment (jleveque) [reply config/main.py]: Changed.

File: config/main.py (hunk 2 above)
Comment (jleveque) [reply config/main.py]: Changed.

File: config/main.py (hunk 3 above)
Comment (jleveque) [reply config/main.py]: Changed.

File: debug/main.py (hunk 5 above)
Comment (jleveque) [reply debug/main.py]: Changed. Also changed in undebug/main.py.

File: debug/main.py (hunk 4 above)
Comment (jleveque) [reply debug/main.py]: Changed. Also changed in undebug/main.py.

File: pddf_fanutil/main.py (hunk 6 above)
Comment (jleveque) [reply pddf_fanutil/main.py]: Changed.

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
@lguohan: I addressed your comments last week. Please review again.

File: undebug/main.py [hunk 7]
```diff
@@ -97,7 +97,7 @@ def pbr(additional):
         command += '"'
         run_command(command)
 
-    @bgp.command()
+    @bgp.command('update_groups')
```
Comment (lguohan) [inline undebug/main.py:100]: update-groups

File: debug/main.py [hunk 8]
```diff
@@ -97,7 +97,7 @@ def pbr(additional):
         command += '"'
         run_command(command)
 
-    @bgp.command()
+    @bgp.command('update_groups')
```
Comment (lguohan) [inline debug/main.py:100]: update-groups

File: undebug/main.py (hunk 7 above)
Comment (jleveque) [reply undebug/main.py]: Done in latest commit.

File: debug/main.py (hunk 8 above)
Comment (jleveque) [reply debug/main.py]: Done in latest commit.

**Comment by abdosi** [comment]
@jleveque Please Create PR 201911. Cherry-pick is not clean

**Comment by jleveque** [comment]
@abdosi: Done. 201911 PR here: https://github.com/Azure/sonic-utilities/pull/868

**Pull request merged**